core/templates/components/skill-selector/select-skill-modal.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { FormsModule } from '@angular/forms';
+import { MatCardModule } from '@angular/material/card';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatRadioModule } from '@angular/material/radio';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { SelectSkillModalComponent } from './select-skill-modal.component';
+import { SkillSelectorComponent } from './skill-selector.component';
+import { SkillsCategorizedByTopics } from 'pages/topics-and-skills-dashboard-page/skills-list/skills-list.component';
+import { ShortSkillSummary, ShortSkillSummaryBackendDict } from 'domain/skill/short-skill-summary.model';
+import { SkillSummary, SkillSummaryBackendDict } from 'domain/skill/skill-summary.model';
+
+
+describe('Select Skill Modal', () => {
+  let fixture: ComponentFixture<SelectSkillModalComponent>;
+  let componentInstance: SelectSkillModalComponent;
+  let ngbActiveModal: NgbActiveModal;
+  let allowSkillsFromOtherTopics: boolean;
+  let shortSkillSummaryBackendDict: ShortSkillSummaryBackendDict = {
+    skill_id: '1',
+    skill_description: 'description1'
+  };
+  let shortSkillSummary: ShortSkillSummary = (
+    ShortSkillSummary.createFromBackendDict(shortSkillSummaryBackendDict));
+  let categorizedSkills: SkillsCategorizedByTopics = {
+    'Dummy Topic': {
+      Subtopic1: [shortSkillSummary]
+    }
+  };
+  let skillsInSameTopicCount: number = 3;
+  let skillSummaryBackendDict: SkillSummaryBackendDict = {
+    id: '3',
+    description: 'description3',
+    language_code: 'language_code',
+    version: 1,
+    misconception_count: null,
+    worked_examples_count: null,
+    skill_model_created_on: 2,
+    skill_model_last_updated: 3
+  };
+  let untriagedSkillSummaries: SkillSummary[] = (
+    [SkillSummary.createFromBackendDict(skillSummaryBackendDict)]);
+  let skillSummaries: SkillSummaryBackendDict[] = [skillSummaryBackendDict];
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        MatCardModule,
+        MatRadioModule,
+        MatCheckboxModule,
+        FormsModule
+      ],
+      declarations: [
+        SelectSkillModalComponent,
+        SkillSelectorComponent
+      ],
+      providers: [
+        NgbActiveModal
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SelectSkillModalComponent);
+    componentInstance = fixture.componentInstance;
+    componentInstance.allowSkillsFromOtherTopics = allowSkillsFromOtherTopics;
+    componentInstance.categorizedSkills = categorizedSkills;
+    componentInstance.skillsInSameTopicCount = skillsInSameTopicCount;
+    componentInstance.skillSummaries = skillSummaries;
+    componentInstance.untriagedSkillSummaries = untriagedSkillSummaries;
+    ngbActiveModal = TestBed.inject(NgbActiveModal);
+    ngbActiveModal = (ngbActiveModal as unknown) as
+      jasmine.SpyObj<NgbActiveModal>;
```

Comment:
Why is this required. If in case it is required, why is it not done in the previous line?

Reply:
I added this here by mistake. Removed!